x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CachedBlobContainerIndexInput.java

```diff
@@ -224,31 +235,33 @@ public void prefetchPart(final int part) throws IOException {
                     while (remainingBytes > 0L) {
                         assert totalBytesRead + remainingBytes == rangeLength;
                         final int bytesRead = readSafe(input, copyBuffer, rangeStart, rangeEnd, remainingBytes, cacheFileReference);
+
                         final long readStart = rangeStart + totalBytesRead;
-                        cacheFile.fetchRange(readStart, readStart + bytesRead, (start, end) -> {
-                            logger.trace(
-                                "prefetchPart: range [{}-{}] of file [{}] is available in cache",
-                                start,
-                                end,
-                                fileInfo.physicalName()
-                            );
-                            return Math.toIntExact(end - start);
-                        }, (start, end) -> {
-                            final ByteBuffer byteBuffer = ByteBuffer.wrap(
-                                copyBuffer,
-                                Math.toIntExact(start - readStart),
-                                Math.toIntExact(end - start)
-                            );
-                            final int writtenBytes = positionalWrite(fc, start, byteBuffer);
-                            logger.trace(
-                                "prefetchPart: writing range [{}-{}] of file [{}], [{}] bytes written",
-                                start,
-                                end,
-                                fileInfo.physicalName(),
-                                writtenBytes
-                            );
-                            totalBytesWritten.addAndGet(writtenBytes);
-                        });
+                        final long readEnd = readStart + bytesRead;
+
+                        cacheFile.fetchAsync(
+                            Tuple.tuple(readStart, readEnd),
+                            Tuple.tuple(readStart, readStart),
```

Comment:
Maybe it deserves some comment: we don't care about reading the cached data for prewarming, it just need to cache the range. This is why it got passed an empty range to read. This way if the range is already (or about to be) written by a concurrent search it returns immediately and process the next small range. If the range is not available then the cache data will be written by this prewarming task.